buildscripts/Set-VersionInfo.ps1

```diff
@@ -0,0 +1,58 @@
+<#
+
+.SYNOPSIS
+
+Extracts build version information from Settings.proj and the
+environment. Then constructs several flavor version strings and
+uses them, along with the current year to
+
+  - update buildscripts/CommonAssemblyInfo.cs,
+  - update all project.json files, and
+  - emit TeamCity directives so it can know about the values
+
+#>
+
+[xml] $settings = Get-Content 'Settings.proj'
```

Comment:
As you can see, I didn't move the canonical version number components from `Settings.proj`. If you've a preferred place, I could do so. I considered the `AssemblyFileVersion` portion of `CommonAssemblyInfo.cs`, but thought that editing that value near all the `0.0.0`s might be confusing.